src/RestApi/StoreApi/Schemas/OrderItemSchema.php

```diff
@@ -0,0 +1,280 @@
+	 * Convert a WooCommerce cart item to an object suitable for the response.
+	 *
+	 * @todo Variation is stored to meta - how can we gather for response?
+	 *
+	 * @param \WC_Order_Item_Product $line_item Order line item array.
+	 * @return array
+	 */
+	public function get_item_response( $line_item ) {
+		$product = $line_item->get_product();
+
+		return [
+			'id'        => $line_item->get_variation_id() ? $line_item->get_variation_id() : $line_item->get_product_id(),
+			'quantity'  => $line_item->get_quantity(),
+			'name'      => $product ? $product->get_title() : null,
+			'sku'       => $product ? $product->get_sku() : null,
+			'permalink' => $product ? $product->get_permalink() : null,
+			'images'    => $product ? ( new ProductImages() )->images_to_array( $product ) : null,
+			'variation' => $this->format_variation_data( $line_item, $product ),
+			'totals'    => array_merge(
+				$this->get_store_currency_response(),
+				[
+					'line_subtotal'     => $this->prepare_money_response( $line_item->get_subtotal(), wc_get_price_decimals() ),
+					'line_subtotal_tax' => $this->prepare_money_response( $line_item->get_subtotal_tax(), wc_get_price_decimals() ),
+					'line_total'        => $this->prepare_money_response( $line_item->get_total(), wc_get_price_decimals() ),
+					'line_total_tax'    => $this->prepare_money_response( $line_item->get_total_tax(), wc_get_price_decimals() ),
+				]
+			),
+		];
+	}
+
+	/**
+	 * Format variation data. For line items we get meta data and format it.
+	 *
+	 * @param \WC_Order_Item_Product $line_item Line item from the order.
+	 * @param \WC_Product            $product Product data.
+	 * @return array
+	 */
+	protected function format_variation_data( $line_item, $product ) {
+		$return         = [];
+		$line_item_meta = $line_item->get_meta_data();
+		$attribute_keys = array_keys( $product->get_attributes() );
+
+		foreach ( $line_item_meta as $meta ) {
+			$key   = $meta->key;
+			$value = $meta->value;
+
+			if ( ! in_array( $key, $attribute_keys, true ) ) {
+				continue;
+			}
+
+			$taxonomy = wc_attribute_taxonomy_name( str_replace( 'pa_', '', urldecode( $key ) ) );
+
+			if ( taxonomy_exists( $taxonomy ) ) {
+				// If this is a term slug, get the term's nice name.
+				$term = get_term_by( 'slug', $value, $taxonomy );
+				if ( ! is_wp_error( $term ) && $term && $term->name ) {
+					$value = $term->name;
+				}
+				$label = wc_attribute_label( $taxonomy );
+			} else {
+				// If this is a custom option slug, get the options name.
+				$value = apply_filters( 'woocommerce_variation_option_name', $value, null, $taxonomy, $product );
+				$label = wc_attribute_label( $name, $product );
+			}
+
+			$return[ $label ] = $value;
+		}
+
+		return $return;
+	}
+
+	/**
+	 * Get product attribute taxonomy name.
+	 *
+	 * @param string      $slug   Taxonomy name.
+	 * @param \WC_Product $object Product data.
+	 * @return string
+	 */
+	protected function get_attribute_taxonomy_name( $slug, $object ) {
```

Comment:
Where is this method called.  I did a search in the blocks plugin and I see it defined in the `CartItemSchema` class two but don't see it called anywhere?

Also if `$object` is always expected to be an instance of `\WC_Product`, let's typehint that and maybe name it `$product`?